### ISSUE TITLE
dtest regressions due to uuid sstable generations

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylladb#14283]
Seen in https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/
With scylladb/scylladb@d1dc579062fb599e0e0cee75e8592f17b2a1884d

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/compaction_additional_test/TestCompactionAdditional/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split001___test_compact_tombstones_when_memtable_flush_one_node_stopped/
```
AssertionError: Tombstones were not found
assert False
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/compaction_additional_test/TestTimeWindowDataSegregation/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split001___test_compaction_remove_deleted_rows_in_previous_time_window/
```
ccmlib.node.NodeError: File doesn't seem to be a valid sstable filename: /jenkins/workspace/scylla-master/dtest-daily-release/scylla/.dtest/dtest-sdttjcpz/test/node1/data/ks/test-e50051000dbc11ee9e3b2281802c4cc4/me-3g6x_0r4m_0dpts2emi57wz3y1tw-big-Data.db
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/compaction_test/TestCompaction/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split001___test_compaction_delete_2_LeveledCompactionStrategy_/
(with all compaction strategies)
```
AssertionError: Error: expected 10 deleted partitions but found 0:
  
assert 0 == 10
  +0
  -10
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/resharding_test/TestReshardingTombstonesSingleNode/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split001___test_disable_tombstone_removal_during_reshard_1_LeveledCompactionStrategy_15_/
(in all variations)
```
AssertionError: All tombstones were removed
assert 0 > 0
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/offline_tools_test/TestOfflineTools/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split001___test_sstablelevelreset/
```
AssertionError: Message was not found in stdout
assert 'Found no sstables, did you give the correct keyspace' in ''
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/compaction_test/TestCompaction/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split007___test_compaction_delete_tombstone_gc_repair_LeveledCompactionStrategy_/
(in all variants)
```
AssertionError: Error: expected 10 deleted partitions but found 0:
  
assert 0 == 10
  +0
  -10
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/offline_tools_test/TestOfflineTools/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split007___test_sstableverify/
```
AssertionError: Invalid exit code: 1
  stdout:
  
  stderr:
  Unknown keyspace/table keyspace1.standard1
  
assert 1 == 0
  +1
  -0
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/sstable_generation_loading_test/TestSSTableGenerationAndLoading/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split009___test_sstableloader_case_sensitive_with_quotas/
```
AssertionError: The stderr of sstableloader has errors: WARN  10:17:12,119 Skipping file me-3g6x_0sic_4r1eo23d0mkrb3fs2l-big-Index.db - unsupported SSTable format or version
  WARN  10:17:12,121 Skipping file me-3g6x_0sic_4r1eo23d0mkrb3fs2l-big-Data.db - unsupported SSTable format or version
  WARN  10:17:12,121 Skipping file me-3g6x_0sic_4r94g2wi9ly0ysp8f1-big-Index.db - unsupported SSTable format or version
  WARN  10:17:12,121 Skipping file me-3g6x_0sic_4r94g2wi9ly0ysp8f1-big-Data.db - unsupported SSTable format or version
  WARN  10:17:12,121 Skipping file me-3g6x_0sic_4r1eo23d0mkrb3fs2l-big-Digest.crc32 - unsupported SSTable format or version
  WARN  10:17:12,121 Skipping file me-3g6x_0sic_4r94g2wi9ly0ysp8f1-big-Digest.crc32 - unsupported SSTable format or version
  WARN  10:17:12,121 Skipping file me-3g6x_0sic_4r1eo23d0mkrb3fs2l-big-Summary.db - unsupported SSTable format or version
  WARN  10:17:12,121 Skipping file me-3g6x_0sic_4r94g2wi9ly0ysp8f1-big-Summary.db - unsupported SSTable format or version
  WARN  10:17:12,122 Skipping file me-3g6x_0sic_4r1eo23d0mkrb3fs2l-big-Filter.db - unsupported SSTable format or version
  WARN  10:17:12,122 Skipping file me-3g6x_0sic_4r94g2wi9ly0ysp8f1-big-Filter.db - unsupported SSTable format or version
  WARN  10:17:12,122 Skipping file me-3g6x_0sic_4r1eo23d0mkrb3fs2l-big-Statistics.db - unsupported SSTable format or version
  WARN  10:17:12,122 Skipping file me-3g6x_0sic_4r1eo23d0mkrb3fs2l-big-CompressionInfo.db - unsupported SSTable format or version
  WARN  10:17:12,122 Skipping file me-3g6x_0sic_4r94g2wi9ly0ysp8f1-big-Statistics.db - unsupported SSTable format or version
  WARN  10:17:12,122 Skipping file me-3g6x_0sic_4r1eo23d0mkrb3fs2l-big-Scylla.db - unsupported SSTable format or version
  WARN  10:17:12,122 Skipping file me-3g6x_0sic_4r94g2wi9ly0ysp8f1-big-CompressionInfo.db - unsupported SSTable format or version
  WARN  10:17:12,122 Skipping file me-3g6x_0sic_4r1eo23d0mkrb3fs2l-big-TOC.txt - unsupported SSTable format or version
  WARN  10:17:12,122 Skipping file me-3g6x_0sic_4r94g2wi9ly0ysp8f1-big-Scylla.db - unsupported SSTable format or version
  WARN  10:17:12,122 Skipping file me-3g6x_0sic_4r94g2wi9ly0ysp8f1-big-TOC.txt - unsupported SSTable format or version
  
assert not True
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/sstable_generation_loading_test/TestSSTableGenerationAndLoading/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split008___test_sstableloader_compression_None_Snappy_/
(in all variants)
```
AssertionError: The stderr of sstableloader has errors: WARN  10:08:44,402 Skipping file me-3g6x_0s4a_0k5b41ydko9us4dqdk-big-Index.db - unsupported SSTable format or version
  WARN  10:08:44,404 Skipping file me-3g6x_0s4a_0k5b42ep7nfglmd32g-big-Index.db - unsupported SSTable format or version
  WARN  10:08:44,404 Skipping file me-3g6x_0s4a_0k5b41ydko9us4dqdk-big-Data.db - unsupported SSTable format or version
  WARN  10:08:44,404 Skipping file me-3g6x_0s4a_0k5b42ep7nfglmd32g-big-Data.db - unsupported SSTable format or version
  WARN  10:08:44,404 Skipping file me-3g6x_0s4a_0k5b42ep7nfglmd32g-big-Digest.crc32 - unsupported SSTable format or version
  WARN  10:08:44,404 Skipping file me-3g6x_0s4a_0k5b41ydko9us4dqdk-big-Digest.crc32 - unsupported SSTable format or version
  WARN  10:08:44,404 Skipping file me-3g6x_0s4a_0k5b42ep7nfglmd32g-big-CRC.db - unsupported SSTable format or version
  WARN  10:08:44,404 Skipping file me-3g6x_0s4a_0k5b41ydko9us4dqdk-big-CRC.db - unsupported SSTable format or version
  WARN  10:08:44,404 Skipping file me-3g6x_0s4a_0k5b42ep7nfglmd32g-big-Summary.db - unsupported SSTable format or version
  WARN  10:08:44,404 Skipping file me-3g6x_0s4a_0k5b41ydko9us4dqdk-big-Summary.db - unsupported SSTable format or version
  WARN  10:08:44,404 Skipping file me-3g6x_0s4a_0k5b42ep7nfglmd32g-big-Filter.db - unsupported SSTable format or version
  WARN  10:08:44,404 Skipping file me-3g6x_0s4a_0k5b41ydko9us4dqdk-big-Filter.db - unsupported SSTable format or version
  WARN  10:08:44,404 Skipping file me-3g6x_0s4a_0k5b42ep7nfglmd32g-big-Statistics.db - unsupported SSTable format or version
  WARN  10:08:44,405 Skipping file me-3g6x_0s4a_0k5b41ydko9us4dqdk-big-Statistics.db - unsupported SSTable format or version
  WARN  10:08:44,405 Skipping file me-3g6x_0s4a_0k5b42ep7nfglmd32g-big-Scylla.db - unsupported SSTable format or version
  WARN  10:08:44,405 Skipping file me-3g6x_0s4a_0k5b41ydko9us4dqdk-big-Scylla.db - unsupported SSTable format or version
  WARN  10:08:44,405 Skipping file me-3g6x_0s4a_0k5b42ep7nfglmd32g-big-TOC.txt - unsupported SSTable format or version
  WARN  10:08:44,405 Skipping file me-3g6x_0s4a_0k5b41ydko9us4dqdk-big-TOC.txt - unsupported SSTable format or version
  
assert not True
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/cql_additional_tests/TestCQL/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split011___test_mc_sstables_case_sensitive_insert/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/cql_additional_tests/TestCQL/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split011___test_mc_sstables_case_sensitive_delete_value/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/cql_additional_tests/TestCQL/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split011___test_mc_sstables_case_sensitive_add_column/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/cql_additional_tests/TestCQL/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split011___test_mc_sstables_case_sensitive_update_value/
```
>       self.mc_migrate_scylla_to_cassandra(keyspace_name=keyspace_name, table_name=table_name, dataset=dataset,
                                            data_amount=data_amount, request=request)

cql_additional_tests.py:6206: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cql_additional_tests.py:6184: in mc_migrate_scylla_to_cassandra
    self.mc_validate_data(session=cassandra_session, table_name=table_name, data_amount=data_amount, dataset=dataset,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <cql_additional_tests.TestCQL object at 0x7f075951d990>
session = <cassandra.cluster.Session object at 0x7f07591ac990>
table_name = '"Columnfamily_For_Mc_Sstables"', data_amount = 10
dataset = [(0, 0, 0, 221115356), (1, 1, 1, 94610143), (2, 2, 2, 101374997), (3, 3, 3, 548513), (4, 4, 4, 35196311), (5, 5, 5, 32576340), ...]
columns = ['"ID"', '"Ck1"', '"cK2"', '"Columnfamily_for_mc_sstables_column1"']
keys_columns_amount = 3

    def mc_validate_data(self, session, table_name, data_amount, dataset,
                         columns=['"ID"', '"Ck1"', '"cK2"', '"Columnfamily_for_mc_sstables_column1"'],
                         keys_columns_amount=3):
        res = list(session.execute('select count(*) from {}'.format(table_name)))
>       assert res[0].count == data_amount
E       assert 0 == 10
E         +0
E         -10

cql_additional_tests.py:6330: AssertionError
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/rolling_upgrade_test/TestRollingUpgrade/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split020___test_rolling_upgrade/
```
AssertionError: expected all table format to be the same found set()
assert 0 == 1
  +0
  -1
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/compaction_additional_test/TestCompactionAdditional/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_compact_tombstones_when_memtable_flush_one_node/
```
AssertionError: Tombstones were not found
assert False
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/compaction_additional_test/TestCompactionAdditionalStrategy/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_compaction_is_started_on_boot_LeveledCompactionStrategy_/
(in all variants)
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/compaction_additional_test/TestTimeWindowDataSegregation/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split021___test_twcs_multiple_sstables_during_decommission/
```
ValueError: invalid literal for int() with base 10: '3g6x_0rd1_18d812e0uqu4xc6z67'
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/snapshot_test/TestSchemaFileInSnapshot/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_restore_snapshot_by_table_schema_file_with_sstableloader/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/snapshot_test/TestSchemaFileInSnapshot/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_restoring_by_schema_with_mv_use_sstableloader/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/snapshot_test/TestSchemaFileInSnapshot/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_restoring_by_schema_with_mv_use_sstableloader_multiple_tables/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/snapshot_test/TestSchemaFileInSnapshot/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_restoring_snapshot_with_indexes_use_sstableloader/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/snapshot_test/TestSchemaFileInSnapshot/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_restoring_snapshot_with_indexes_use_sstableloader_multiple_tables/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/snapshot_test/TestSchemaFileInSnapshot/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_restore_data_for_all_native_data_types_from_snapshot_with_sstablesloader/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/snapshot_test/TestSchemaFileInSnapshot/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_restore_data_from_snapshot_with_frozen_udt_with_sstablesloader/
```
self = <snapshot_test.TestSchemaFileInSnapshot object at 0x7f326c142e50>

    def test_restore_snapshot_by_table_schema_file_with_sstableloader(self):
>       self.create_restore_data_with_snapshot(use_sstableloader=True)

snapshot_test.py:1037: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
snapshot_test.py:1200: in create_restore_data_with_snapshot
    self.check_rows_number_in_table(session, "ks", "cf", 100)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <snapshot_test.TestSchemaFileInSnapshot object at 0x7f326c142e50>
session = <cassandra.cluster.Session object at 0x7f326c15e890>, ks = 'ks'
cf = 'cf', number = 100

    def check_rows_number_in_table(self, session, ks, cf, number):
        rows = session.execute(f'SELECT count(*) from {ks}.{cf}')
>       assert rows.one()[0] == number
E       assert 0 == 100
E         +0
E         -100

snapshot_test.py:1532: AssertionError
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/scrub_test/TestScrubIndexes/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_scrub_static_table/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/scrub_test/TestScrubIndexes/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_scrub_collections_table_without_indexes/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/scrub_test/TestScrub/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_nodetool_scrub/
```
AssertionError: List of sstables before and after scrub differs
...
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/sstableloader_test/TestAdditionalTestSSTableLoader/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_ignore_missing_one_column_by_drop_table_from_backup/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/sstableloader_test/TestAdditionalTestSSTableLoader/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_ignore_missing_one_column_by_drop_column_from_backup/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/sstableloader_test/TestAdditionalTestSSTableLoader/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_ignore_missing_two_column_by_drop_column_from_backup/
```
Exception: sstableloader command failed, exit status: 0,

STDERR: b'WARN  10:19:35,091 Skipping file me-3g6x_0soj_2fvkw2q9h0reka9oh0-big-Index.db - unsupported SSTable format or version\nWARN  10:19:35,093 Skipping file me-3g6x_0soj_2fvkw2q9h0reka9oh0-big-Data.db - unsupported SSTable format or version\nWARN  10:19:35,093 Skipping file me-3g6x_0soj_2fvkw2x1d6jfrruexw-big-Index.db - unsupported SSTable format or version\nWARN  10:19:35,093 Skipping file me-3g6x_0soj_2fvkw2x1d6jfrruexw-big-Data.db - unsupported SSTable format or version\nWARN  10:19:35,093 Skipping file me-3g6x_0soj_2fvkw2q9h0reka9oh0-big-Digest.crc32 - unsupported SSTable format or version\nWARN  10:19:35,093 Skipping file me-3g6x_0soj_2fvkw2x1d6jfrruexw-big-Digest.crc32 - unsupported SSTable format or version\nWARN  10:19:35,093 Skipping file me-3g6x_0soj_2fvkw2q9h0reka9oh0-big-Summary.db - unsupported SSTable format or version\nWARN  10:19:35,093 Skipping file me-3g6x_0soj_2fvkw2q9h0reka9oh0-big-Filter.db - unsupported SSTable format or version\nWARN  10:19:35,093 Skipping file me-3g6x_0soj_2fvkw2x1d6jfrruexw-big-Summary.db - unsupported SSTable format or version\nWARN  10:19:35,093 Skipping file me-3g6x_0soj_2fvkw2q9h0reka9oh0-big-Statistics.db - unsupported SSTable format or version\nWARN  10:19:35,093 Skipping file me-3g6x_0soj_2fvkw2x1d6jfrruexw-big-Filter.db - unsupported SSTable format or version\nWARN  10:19:35,094 Skipping file me-3g6x_0soj_2fvkw2q9h0reka9oh0-big-CompressionInfo.db - unsupported SSTable format or version\nWARN  10:19:35,094 Skipping file me-3g6x_0soj_2fvkw2x1d6jfrruexw-big-Statistics.db - unsupported SSTable format or version\nWARN  10:19:35,094 Skipping file me-3g6x_0soj_2fvkw2x1d6jfrruexw-big-CompressionInfo.db - unsupported SSTable format or version\nWARN  10:19:35,094 Skipping file me-3g6x_0soj_2fvkw2q9h0reka9oh0-big-Scylla.db - unsupported SSTable format or version\nWARN  10:19:35,094 Skipping file me-3g6x_0soj_2fvkw2x1d6jfrruexw-big-Scylla.db - unsupported SSTable format or version\nWARN  10:19:35,094 Skipping file me-3g6x_0soj_2fvkw2q9h0reka9oh0-big-TOC.txt - unsupported SSTable format or version\nWARN  10:19:35,094 Skipping file me-3g6x_0soj_2fvkw2x1d6jfrruexw-big-TOC.txt - unsupported SSTable format or version\n'
,
STDOUT: b'===== Using optimized driver!!! =====\n  0% done.        0 statements sent (in        0 batches,        0 failed).\r\n       0 statements generated.\n       0 cql rows processed in        0 partitions.\n       0 cql rows and        0 partitions deleted.\n       0 local and        0 remote counter shards where skipped.\n'
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/compaction_additional_test/TestTimeWindowDataSegregation/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split021___test_enable_disable_optimized_query_for_twcs/
```
AssertionError: iterable_item_added: 750, first 5 differences:
  {'root[94620]': {'ck': 2850, 'pk': -1893954442, 'v': b'a'},
   'root[94621]': {'ck': 2851, 'pk': -1893954442, 'v': b'a'},
   'root[94622]': {'ck': 2852, 'pk': -1893954442, 'v': b'a'},
   'root[94623]': {'ck': 2853, 'pk': -1893954442, 'v': b'a'},
   'root[94624]': {'ck': 2854, 'pk': -1893954442, 'v': b'a'}}
assert not {'root[94620]': {'ck': 2850, 'pk': -1893954442, 'v': b'a'}, 'root[94621]': {'ck': 2851, 'pk': -1893954442, 'v': b'a'},...ot[94622]': {'ck': 2852, 'pk': -1893954442, 'v': b'a'}, 'root[94623]': {'ck': 2853, 'pk': -1893954442, 'v': b'a'}, ...}
 +  where {'root[94620]': {'ck': 2850, 'pk': -1893954442, 'v': b'a'}, 'root[94621]': {'ck': 2851, 'pk': -1893954442, 'v': b'a'},...ot[94622]': {'ck': 2852, 'pk': -1893954442, 'v': b'a'}, 'root[94623]': {'ck': 2853, 'pk': -1893954442, 'v': b'a'}, ...} = <built-in method get of DeepDiff object at 0x7fcf81587650>('iterable_item_added')
 +    where <built-in method get of DeepDiff object at 0x7fcf81587650> = {'values_changed': {"root[0]['ck']": {'new_value': 421, 'old_value': 436}, "root[1]['ck']": {'new_value': 422, 'old_va... 'root[95368]': {'pk': -1893954442, 'ck': 3598, 'v': b'a'}, 'root[95369]': {'pk': -1893954442, 'ck': 3599, 'v': b'a'}}}.get
```

**Comment by bhalevy** [comment]
@scylladb/scylla-maint let's revert https://github.com/scylladb/scylladb/commit/d1dc579062fb599e0e0cee75e8592f17b2a1884d as a fix for this issue.

**Comment by bhalevy** [comment]
@tchaikov we need to make sure there are no regressions in advance to re-merging https://github.com/scylladb/scylladb/pull/13932

**Comment by denesb** [comment]
Revert queued as 562087beff150b4a785d0305b7310cc982b465cb.

**Comment by fruch** [comment]
can confirm it's also breaking some SCT nemesis `disrupt_destroy_data_then_repair`, again test code that assumes sstable versions are numbers.

**Comment by tchaikov** [comment]
@fruch could you please point me to the failing tests? so i can try to fix it.

**Comment by fruch** [comment]
> @fruch could you please point me to the failing tests? so i can try to fix it.

test failure:
https://argus.scylladb.com/test/aa05fe67-89bd-4e30-88db-5e2b2e3d986d/runs?additionalRuns[]=7f44828b-1641-4080-b990-8cbeaf3eaafa

it's basically this function that assume the structure of the file name, so it can delete a whole version as once:
https://github.com/scylladb/scylla-cluster-tests/blame/dce040c3e5e1a77245c9c7910277fc50668b4183/sdcm/nemesis.py#L994

**Comment by tchaikov** [comment]
thank you @fruch !

**Comment by tchaikov** [comment]
> https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/compaction_additional_test/TestTimeWindowDataSegregation/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split021___test_enable_disable_optimized_query_for_twcs/
> 
> ```
> AssertionError: iterable_item_added: 750, first 5 differences:
>   {'root[94620]': {'ck': 2850, 'pk': -1893954442, 'v': b'a'},
>    'root[94621]': {'ck': 2851, 'pk': -1893954442, 'v': b'a'},
>    'root[94622]': {'ck': 2852, 'pk': -1893954442, 'v': b'a'},
>    'root[94623]': {'ck': 2853, 'pk': -1893954442, 'v': b'a'},
>    'root[94624]': {'ck': 2854, 'pk': -1893954442, 'v': b'a'}}
> assert not {'root[94620]': {'ck': 2850, 'pk': -1893954442, 'v': b'a'}, 'root[94621]': {'ck': 2851, 'pk': -1893954442, 'v': b'a'},...ot[94622]': {'ck': 2852, 'pk': -1893954442, 'v': b'a'}, 'root[94623]': {'ck': 2853, 'pk': -1893954442, 'v': b'a'}, ...}
>  +  where {'root[94620]': {'ck': 2850, 'pk': -1893954442, 'v': b'a'}, 'root[94621]': {'ck': 2851, 'pk': -1893954442, 'v': b'a'},...ot[94622]': {'ck': 2852, 'pk': -1893954442, 'v': b'a'}, 'root[94623]': {'ck': 2853, 'pk': -1893954442, 'v': b'a'}, ...} = <built-in method get of DeepDiff object at 0x7fcf81587650>('iterable_item_added')
>  +    where <built-in method get of DeepDiff object at 0x7fcf81587650> = {'values_changed': {"root[0]['ck']": {'new_value': 421, 'old_value': 436}, "root[1]['ck']": {'new_value': 422, 'old_va... 'root[95368]': {'pk': -1893954442, 'ck': 3598, 'v': b'a'}, 'root[95369]': {'pk': -1893954442, 'ck': 3599, 'v': b'a'}}}.get
> ```

i think this one is a known issue tracked by #11861

**Comment by bhalevy** [comment]
> @fruch could you please point me to the failing tests? so i can try to fix it.

@tchaikov I tried listing all the dtests that failed due to sstable uuid generation in the description of this issue.

**Comment by bhalevy** [comment]
> > https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/274/testReport/compaction_additional_test/TestTimeWindowDataSegregation/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split021___test_enable_disable_optimized_query_for_twcs/
> > ```
> > AssertionError: iterable_item_added: 750, first 5 differences:
> >   {'root[94620]': {'ck': 2850, 'pk': -1893954442, 'v': b'a'},
> >    'root[94621]': {'ck': 2851, 'pk': -1893954442, 'v': b'a'},
> >    'root[94622]': {'ck': 2852, 'pk': -1893954442, 'v': b'a'},
> >    'root[94623]': {'ck': 2853, 'pk': -1893954442, 'v': b'a'},
> >    'root[94624]': {'ck': 2854, 'pk': -1893954442, 'v': b'a'}}
> > assert not {'root[94620]': {'ck': 2850, 'pk': -1893954442, 'v': b'a'}, 'root[94621]': {'ck': 2851, 'pk': -1893954442, 'v': b'a'},...ot[94622]': {'ck': 2852, 'pk': -1893954442, 'v': b'a'}, 'root[94623]': {'ck': 2853, 'pk': -1893954442, 'v': b'a'}, ...}
> >  +  where {'root[94620]': {'ck': 2850, 'pk': -1893954442, 'v': b'a'}, 'root[94621]': {'ck': 2851, 'pk': -1893954442, 'v': b'a'},...ot[94622]': {'ck': 2852, 'pk': -1893954442, 'v': b'a'}, 'root[94623]': {'ck': 2853, 'pk': -1893954442, 'v': b'a'}, ...} = <built-in method get of DeepDiff object at 0x7fcf81587650>('iterable_item_added')
> >  +    where <built-in method get of DeepDiff object at 0x7fcf81587650> = {'values_changed': {"root[0]['ck']": {'new_value': 421, 'old_value': 436}, "root[1]['ck']": {'new_value': 422, 'old_va... 'root[95368]': {'pk': -1893954442, 'ck': 3598, 'v': b'a'}, 'root[95369]': {'pk': -1893954442, 'ck': 3599, 'v': b'a'}}}.get
> > ```
> 
> i think this one is a known issue tracked by #11861

yes

**Comment by tchaikov** [comment]
> > @fruch could you please point me to the failing tests? so i can try to fix it.
> 
> @tchaikov I tried listing all the dtests that failed due to sstable uuid generation in the description of this issue.

@bhalevy hi Benny, thank you for collecting them! really appreciate this. i was asking Israel for the sct failure. and it should be fixed by https://github.com/scylladb/scylla-cluster-tests/pull/6263

**Comment by bhalevy** [comment]
> > > @fruch could you please point me to the failing tests? so i can try to fix it.
> > 
> > 
> > @tchaikov I tried listing all the dtests that failed due to sstable uuid generation in the description of this issue.
> 
> @bhalevy hi Benny, thank you for collecting them! really appreciate this. i was asking Israel for the sct failure. and it should be fixed by [scylladb/scylla-cluster-tests#6263](https://github.com/scylladb/scylla-cluster-tests/pull/6263) . the fix on dtest side is at [scylladb/scylla-dtest#3244](https://github.com/scylladb/scylla-dtest/pull/3244)

I guess that none of the failed dtests was included in `next_gating`.
To prevent such a case in the future we should increase the coverage of `next_gating` as much as possible.
Ideally it should include all stable test cases in `dtest_full` maybe except the long running ones.

**Comment by fruch** [comment]
> > > > @fruch could you please point me to the failing tests? so i can try to fix it.
> > > 
> > > 
> > > @tchaikov I tried listing all the dtests that failed due to sstable uuid generation in the description of this issue.
> > 
> > @bhalevy hi Benny, thank you for collecting them! really appreciate this. i was asking Israel for the sct failure. and it should be fixed by [scylladb/scylla-cluster-tests#6263](https://github.com/scylladb/scylla-cluster-tests/pull/6263) . the fix on dtest side is at [scylladb/scylla-dtest#3244](https://github.com/scylladb/scylla-dtest/pull/3244)
> 
> I guess that none of the failed dtests was included in `next_gating`.
> To prevent such a case in the future we should increase the coverage of `next_gating` as much as possible.
> Ideally it should include all stable test cases in `dtest_full` maybe except the long running ones.

Maybe we should add exactly those in preparation for this feature ?

**Comment by bhalevy** [comment]
> > > > > @fruch could you please point me to the failing tests? so i can try to fix it.
> > > > 
> > > > 
> > > > @tchaikov I tried listing all the dtests that failed due to sstable uuid generation in the description of this issue.
> > > 
> > > 
> > > @bhalevy hi Benny, thank you for collecting them! really appreciate this. i was asking Israel for the sct failure. and it should be fixed by [scylladb/scylla-cluster-tests#6263](https://github.com/scylladb/scylla-cluster-tests/pull/6263) . the fix on dtest side is at [scylladb/scylla-dtest#3244](https://github.com/scylladb/scylla-dtest/pull/3244)
> > 
> > 
> > I guess that none of the failed dtests was included in `next_gating`.
> > To prevent such a case in the future we should increase the coverage of `next_gating` as much as possible.
> > Ideally it should include all stable test cases in `dtest_full` maybe except the long running ones.
> 
> Maybe we should add exactly those in preparation for this feature ?

Yes, I think it's a good idea to add all stable dtests that failed here to next gating to prevent regression.

**Comment by bhalevy** [comment]
@tchaikov can this issue be closed?

**Comment by tchaikov** [comment]
> > > > > @fruch could you please point me to the failing tests? so i can try to fix it.
> > > > 
> > > > 
> > > > @tchaikov I tried listing all the dtests that failed due to sstable uuid generation in the description of this issue.
> > > 
> > > 
> > > @bhalevy hi Benny, thank you for collecting them! really appreciate this. i was asking Israel for the sct failure. and it should be fixed by [scylladb/scylla-cluster-tests#6263](https://github.com/scylladb/scylla-cluster-tests/pull/6263) . the fix on dtest side is at [scylladb/scylla-dtest#3244](https://github.com/scylladb/scylla-dtest/pull/3244)
> > 
> > 
> > I guess that none of the failed dtests was included in `next_gating`.
> > To prevent such a case in the future we should increase the coverage of `next_gating` as much as possible.
> > Ideally it should include all stable test cases in `dtest_full` maybe except the long running ones.
> 
> Maybe we should add exactly those in preparation for this feature ?

i wish i had a crystal ball so i could foresee these test failures so i can add them beforehand =(

**Comment by tchaikov** [comment]
> @tchaikov can this issue be closed?

i think so. the regressions were addressed by 562087beff150b4a785d0305b7310cc982b465cb .
